lib/plugins/environments.js

```diff
@@ -147,12 +147,13 @@ module.exports = class Environments extends Diffable {
 
     if (attrs.deployment_branch_policy && attrs.deployment_branch_policy.custom_branches) {
       await Promise.all(
-        attrs.deployment_branch_policy.custom_branches.map(name =>
+        attrs.deployment_branch_policy.custom_branches.map(rule =>
           this.github.request('POST /repos/:org/:repo/environments/:environment_name/deployment-branch-policies', {
             org: this.repo.owner,
             repo: this.repo.repo,
             environment_name: attrs.name,
-            name
+            name: rule.name || rule,
+            type: rule.type || 'branch'
```

Comment:
because of the default values, i think we also need to handle similar in https://github.com/repository-settings/app/blob/36173c5eadad3084dc6b3153b466d834e167a7c5/lib/plugins/environments.js#L54-L59 to avoid repetitively updating rules that are already in an ok state.

Reply:
@travi, thanks for the nice catch.
I'll address this comment this week.

Reply:
@travi, you were right; there was a bug related to repetitively updating.
I made the fix and added unit tests to ensure the bug was solved.
May I have a review?

P.S.: Sorry for the delayed response.